### PR TITLE
Create revision path for new watershed module

### PR DESCRIPTION
**Pull request by gnodar01**
@bethac07 This is an outstanding commit that was going to be included in `library-watershed`, but since that's already been pulled I just created a new PR instead. Only `upgrade_settings` has any important changes, in the 3 -> 4 condition, the rest is just tidying.

### CODE REVIEW

**Comment by bethac07** [comment]
Sorry! I thought you said it was good when we spoke yesterday, my bad

File: cellprofiler/modules/watershed.py [hunk 1]
```diff
@@ -566,31 +573,61 @@ def upgrade_settings(self, setting_values, variable_revision_number, module_name
 
         if variable_revision_number == 1:
             # Last two items were moved down to add more options for seeded watershed
-            __settings__ = setting_values[:-2]
+            new_values = setting_values[:-2]
 
-            # Add default connectivity and compactness
-            __settings__ += [1, 0.0]
+            new_values += [default_settings["connectivity"],
```
Comment (bethac07) [inline cellprofiler/modules/watershed.py:578]: Per previous convention, I thiiiink we want to leave the hardcoding here, but I'm willing to be argued out of it

Comment (bethac07) [reply cellprofiler/modules/watershed.py:578]: Same applies down the rest of the upgrade_settings

File: cellprofiler/modules/watershed.py [hunk 2]
```diff
@@ -258,7 +255,17 @@ def create_settings(self):
...
+Create a 1 pixel wide line around the watershed labels. This effectively
+separates the different objects identified by the watershed algorithm, rather
+than allowing them to touch. The line has the same label as the background.
+""",
+        )
+
+        self.declump_method = Choice(
             text="Declump method",
             choices=[O_SHAPE, O_INTENSITY, O_NONE],
             value=O_SHAPE,
```
Comment (bethac07) [inline cellprofiler/modules/watershed.py:271]: Do we want to use default_settings here too?

Comment (bethac07) [reply cellprofiler/modules/watershed.py:271]: No, I guess not, this is only for hiding advanced, nvm

**Review by bethac07 — APPROVED** [review]

**Comment by gnodar01** [comment]
> Sorry! I thought you said it was good when we spoke yesterday, my bad

No worries, my bad actually, I realized it needed updates after our convo and just forgot to set it to draft. No harm in having a quick follow up PR anyway.

**Pull request merged**